src/style.rs

```diff
@@ -164,6 +164,15 @@ impl ProgressStyle {
         Ok(self)
     }
 
+    /// Specifies that the progress bar is intended to be printed to stderr
+    ///
+    /// The progress bar will determine whether to enable/disable colors based on stderr
+    /// instead of stdout. Under the hood, this uses [`console::colors_enabled_stderr`].
+    pub(crate) fn for_stderr(mut self) -> Self {
+        self.template.set_for_stderr();
```

Comment:
Let's inline `Template::set_for_stderr()` here, and make it take `&mut self` (and return nothing).